src/dotnet/APIView/APIViewWeb/Helpers/CodeFileHelpers.cs

```diff
@@ -1,790 +1,486 @@
 
+using ApiView;
+using APIView;
+using APIView.Model.V2;
 using APIView.TreeToken;
 using APIViewWeb.Extensions;
 using APIViewWeb.LeanModels;
 using System;
-using System.Collections.Concurrent;
 using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
 using System.Linq;
-using System.Threading;
+using System.Reflection.Metadata.Ecma335;
 using System.Threading.Tasks;
 
 namespace APIViewWeb.Helpers
 {
     public class CodeFileHelpers
     {
-        private static int _processorCount = Environment.ProcessorCount;
-        private static SemaphoreSlim _semaphore = new SemaphoreSlim(_processorCount);
-        private static ConcurrentBag<Task> _tasks = new ConcurrentBag<Task>();
-
-
         public static async Task<CodePanelData> GenerateCodePanelDataAsync(CodePanelRawData codePanelRawData)
         {
             var codePanelData = new CodePanelData();
+            var reviewLines = codePanelRawData.activeRevisionCodeFile.ReviewLines;
 
-            for (int idx = 0; idx < codePanelRawData.APIForest.Count; idx++)
-            {
-                var node = codePanelRawData.APIForest[idx];
-                await BuildAPITree(codePanelData: codePanelData, codePanelRawData: codePanelRawData, apiTreeNode: node, 
-                    parentNodeIdHashed: "root", nodePositionAtLevel: idx);
-            };
-
-            if (_processorCount > 1)
+            // Create root node
+            var rootNodeId = "root";
+            if (!codePanelData.NodeMetaDataObj.ContainsKey(rootNodeId))
             {
-                await Task.WhenAll(_tasks);
+                codePanelData.NodeMetaDataObj.TryAdd(rootNodeId, new CodePanelNodeMetaData());
```

Comment:
You could change the models to regular `Dictionaries` since different threads are no longer used.

Reply:
Looks like you already did.